src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2143,33 +2144,61 @@ private void VisitStatementsWithLocalFunctions(BoundBlock block)
             }
         }
 
-        public override BoundNode? VisitLocalFunctionStatement(BoundLocalFunctionStatement localFunc)
+        public override BoundNode? VisitLocalFunctionStatement(BoundLocalFunctionStatement node)
```

Comment:
Alright, so I think speculative analysis in local functions was already broken in current master: we're never calling `snapshotBuilderOpt?.EnterNewWalker()` on the local function symbol, so state saving and restore will be broken.

Reply:
This should reproduce in that variables declared in the local function don't have correct state when restoring for semantic analysis (if previous statements in the local function checked them for null, for example.

---
In reply to: [451908958](https://github.com/dotnet/roslyn/pull/45527#discussion_r451908958) [](ancestors = 451908958)

Reply:
Opened https://github.com/dotnet/roslyn/issues/45825 to track this. I think that in order to not introduce this bug for lambdas in this PR you're going to have to fix this one as well.